apis/workflows/v1alpha1/workflow_types.go

```diff
@@ -44,51 +44,69 @@ type Parent struct {
 type WorkflowsWorkflowSpec struct {
 	Parent `json:",inline"`
-
 	// Description of the workflow provided by the user.
-	// Must be at most 1000 unicode characters long.
+	//  Must be at most 1000 Unicode characters long.
+	//  This is a workflow-wide field and is not tied to a specific revision.
+	// +kcc:proto:field=google.cloud.workflows.v1.Workflow.description
 	Description *string `json:"description,omitempty"`
 
 	// Labels associated with this workflow.
-	// Labels can contain at most 64 entries. Keys and values can be no longer
-	// than 63 characters and can only contain lowercase letters, numeric
-	// characters, underscores, and dashes. Label keys must start with a letter.
-	// International characters are allowed.
-	Labels map[string]string `json:"labels,omitempty"`
+	//  Labels can contain at most 64 entries. Keys and values can be no longer
+	//  than 63 characters and can only contain lowercase letters, numeric
+	//  characters, underscores, and dashes. Label keys must start with a letter.
+	//  International characters are allowed.
+	//  This is a workflow-wide field and is not tied to a specific revision.
+	// +kcc:proto:field=google.cloud.workflows.v1.Workflow.labels
+	// Labels map[string]string `json:"labels,omitempty"`
 
 	// The service account associated with the latest workflow version.
-	// This service account represents the identity of the workflow and determines
-	// what permissions the workflow has.
-	// If not provided, workflow will use the project's default service account.
-	// Modifying this field for an existing workflow results in a new workflow
-	// revision.
+	//  This service account represents the identity of the workflow and determines
+	//  what permissions the workflow has.
+	//  Format: projects/{project}/serviceAccounts/{account} or {account}
+	//
+	//  Using `-` as a wildcard for the `{project}` or not providing one at all
+	//  will infer the project from the account. The `{account}` value can be the
+	//  `email` address or the `unique_id` of the service account.
+	//
+	//  If not provided, workflow will use the project's default service account.
+	//  Modifying this field for an existing workflow results in a new workflow
+	//  revision.
+	// +kcc:proto:field=google.cloud.workflows.v1.Workflow.service_account
 	ServiceAccountRef *refs.IAMServiceAccountRef `json:"serviceAccountRef,omitempty"`
 
-	// Required. Workflow code to be executed. The size limit is 128KB.
-	// +required
+	// Workflow code to be executed. The size limit is 128KB.
+	// +kcc:proto:field=google.cloud.workflows.v1.Workflow.source_contents
 	SourceContents *string `json:"sourceContents,omitempty"`
 
 	// Optional. The resource name of a KMS crypto key used to encrypt or decrypt
 	// the data associated with the workflow.
-	// If not provided, data associated with the workflow will not be
-	// CMEK-encrypted.
-	// +optional
-	KMSCryptoKeyRef *refs.KMSCryptoKeyRef `json:"kmsCryptoKeyRef,omitempty"`
```

Comment:
We need to fix the json name otherwise mapper-generator cannot find out the KRM field.

Reply:
> What if we change the json tag but not the go field name? 

Then we cannot use the auto-mapper functions (but we can write the manual function if we want to use a custom name). This is the auto-mapper if using a different name https://screenshot.googleplex.com/4dhf3w8tMxNcMYz.png

https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/5491 is to update the reference doc about the naming, and auto-mapper  when turning on the generate.sh.

Reply:
Can you try this: 
```
CryptoKeyNameRef *refs.KMSCryptoKeyRef `json:"kmsCryptoKeyRef,omitempty"`
```

Reply:
I misread the original ask. The json and go field does not need to match, but this is alpha, isn't it better to keep them the same? Or is your concern about the crypto_key_"name" conflict with the "ref"?